podcasts/End of Year/Stories/EndOfYearStoriesDataSource.swift

```diff
@@ -2,59 +2,57 @@ import SwiftUI
-        switch storyNumber {
-        case 0:
+        switch stories[storyNumber] {
+        case .intro:
             return IntroStory()
-        case 1:
-            return ListeningTimeStory(listeningTime: listeningTime!)
-        case 2:
-            return ListenedCategoriesStory(listenedCategories: listenedCategories)
-        case 3:
-            return TopListenedCategories(listenedCategories: listenedCategories)
-        case 4:
-            return ListenedNumbersStory(listenedNumbers: listenedNumbers!)
-        case 5:
-            return TopOnePodcastStory(topPodcast: topPodcasts[0])
-        case 6:
-            return TopFivePodcastsStory(podcasts: topPodcasts.map { $0.podcast })
-        case 7:
-            return LongestEpisodeStory(episode: longestEpisode!, podcast: longestEpisode!.parentPodcast()!)
-        default:
+        case .listeningTime:
+            return ListeningTimeStory(listeningTime: data.listeningTime)
+        case .listenedCategories:
+            return ListenedCategoriesStory(listenedCategories: data.listenedCategories)
+        case .topFiveCategories:
+            return TopListenedCategories(listenedCategories: data.listenedCategories)
+        case .listenedNumbers:
+            return ListenedNumbersStory(listenedNumbers: data.listenedNumbers)
+        case .topOnePodcast:
+            return TopOnePodcastStory(topPodcast: data.topPodcasts[0])
+        case .topFivePodcasts:
+            return TopFivePodcastsStory(podcasts: data.topPodcasts.map { $0.podcast })
+        case .longestEpisode:
+            return LongestEpisodeStory(episode: data.longestEpisode, podcast: data.longestEpisodePodcast)
+        case .epilogue:
             return EpilogueStory()
         }
     }
 
     /// The only interactive view we have is the last one, with the replay button
     func interactiveView(for storyNumber: Int) -> AnyView {
-        storyNumber == 8 ? AnyView(EpilogueStory()) : AnyView(EmptyView())
+        switch stories[storyNumber] {
+        case .epilogue:
+            return AnyView(EpilogueStory())
+        default:
+            return AnyView(EmptyView())
+        }
     }
 
     func isReady() async -> Bool {
-        await withCheckedContinuation { continuation in
-            self.listeningTime = DataManager.sharedManager.listeningTime()
-
-            self.listenedCategories = DataManager.sharedManager.listenedCategories()
-
-            self.listenedNumbers = DataManager.sharedManager.listenedNumbers()
-
-            self.topPodcasts = DataManager.sharedManager.topPodcasts()
+        (stories, data) = await EndOfYearStoriesBuilder().build()
 
-            self.longestEpisode = DataManager.sharedManager.longestEpisode()
+        if !stories.isEmpty {
+            stories.insert(.intro, at: 0)
+            stories.append(.epilogue)
 
-            continuation.resume(returning: true)
+            return true
+        } else {
+            return false
         }
```

Comment:
Nit: We don't really need the else here. 

```suggestion
            return true
        }

        return false
```